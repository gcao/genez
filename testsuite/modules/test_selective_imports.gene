#!/usr/bin/env gene run

# Test selective imports
(import "./math.gene" [PI square])
(import "./helpers.gene" [helper => h])

(println "Testing selective imports...")
(println "Direct access to PI:" PI)
(println "Direct access to square:" (square 5))
(println "Aliased helper as h:" (h "test"))

# These should NOT be accessible (not imported)
# (println "E:" E)  # Should fail - not imported
# (println "cube:" (cube 3))  # Should fail - not imported