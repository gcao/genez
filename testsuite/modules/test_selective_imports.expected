Testing selective imports...
Direct access to PI: 3.14159
Direct access to square: 25
Aliased helper as h: test